extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+
+
+def licenses(id):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if id is None:
+        return []
+    else:
+        return [{"license": {
+            "id": id
+        }}]
+
+
+def package_url(node: dict) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    qualifiers = {
+        "prev": node["prev"]
+    }
+    if node["user"]:
+        qualifiers["user"] = node["user"]
+    if node["channel"]:
+        qualifiers["channel"] = node["channel"]
+    if "#" in node["ref"]:
+        qualifiers["rref"] = node["ref"].split("#")[1]
+    qualifiers["repository_url"] = node["remote"] if node["remote"] else "https://center.conan.io"
```

Comment:
Just a minor comment:
> If omitted, https://center.conan.io as default repository is assumed.

You might consider omitting the default value (as it is anyway assumed by the specification).

Reply:
this is true. the purl should be as short as possible.
tools and some OSS-Indexes had problems with long purls. so dropping standard/default values where you can helps shorten the purl.

Reply:
Changed: I set it now only when it is not the default value